opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -296,8 +296,21 @@ namespace Opm
             }
             else {
                 // just write the data to disk
-                writeTimeStepSerial( timer, state, wellState, cellData, substep );
+                int err = 0;
+                try {
+                    writeTimeStepSerial( timer, state, wellState, cellData, substep );
+                    MPI_Bcast(&err, 1, MPI_INT, 0, MPI_COMM_WORLD);
+                } catch (std::runtime_error& msg) {
```

Comment:
I see. So at least it's not a deadlock but that is kinda hard to see. I'll make a suggestion but it may take a little while. I would add the HAVE_MPI though or use the CollectiveCommunication object.